contracts/margin/external/BucketLender/BucketLenderWithRecoveryDelay.sol

```diff
@@ -0,0 +1,95 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+import { Math } from "openzeppelin-solidity/contracts/math/Math.sol";
+import { BucketLender } from "./BucketLender.sol";
+import { MarginHelper } from "../../external/lib/MarginHelper.sol";
+import { MarginCommon } from "../../impl/MarginCommon.sol";
+
+
+/**
+ * @title BucketLenderWithRecoveryDelay
+ * @author dYdX
+ *
+ * Extension of BucketLender that delays the force-recovery time
+ */
+contract BucketLenderWithRecoveryDelay is BucketLender
+{
+    // ============ State Variables ============
+
+    // number of seconds after position has closed that must be waited before force-recovering
+    uint32 public RECOVERY_DELAY;
```

Comment:
why make this a `uint32` and not `uint256`? It doesn't save any space, and I think might cost more gas to shift things when it loads it. Also we're operating on it with other `uint256`'s so it may be more error prone if things need to be casted